File: 1302-delete-characters-to-make-fancy-string/1302-delete-characters-to-make-fancy-string.cpp
class Solution {
public:
    string makeFancyString(string s) {
        int i = 0; 
        string str ="";
        while(i<s.size())
        {
            int j = i+1;
            while(j<s.size() && s[i] == s[j])
            {
                j++;
            }
            if(j-i>=3)
            {
                str.push_back(s[i]);
                str.push_back(s[i]);
                i=j;
            }
            else
            {
                str.push_back(s[i]);
                i++;
            }
        }return str; 
        
    }
};

File: 1302-delete-characters-to-make-fancy-string/1302-delete-characters-to-make-fancy-string.cs
public class Solution {
    public string MakeFancyString(string s) {

        StringBuilder x = new StringBuilder();
        int j =0; int sz = s.Count();
        
        for(int i = 0; i< sz; i++)
        {
            if(j-2>=0 && s[i] == x[j-1] && s[i] == x[j-2])
                continue;
            x.Append(s[i]); j++;
        }return x.ToString();
        
    }
}